src/semiwrap/tool/update_yaml.py

```diff
@@ -0,0 +1,352 @@
+import os
+import pathlib
+import sys
+import traceback
+import tempfile
+import typing as T
+import shutil
+import difflib
+import dictdiffer
+import dictdiffer.utils
+from io import StringIO
+from ruamel.yaml import YAML
+
+from ..autowrap.generator_data import MissingReporter
+from ..cmd.header2dat import make_argparser, generate_wrapper
+from ..makeplan import InputFile, makeplan, BuildTarget, CompilerInfo
+
+
+class YamlUpdater:
+    """
+    This class will parse the headers for a semiwrap project and create or update their corresponding yaml files.
+
+    The process for doing this goes as such:
+        1. Copy backups of the yaml files into a temporary directory
```

Comment:
... so, why does it do this? I think a bunch of things could be simplified if this wasn't the case.

Reply:
Bazel can't edit files in place; they are marked as readonly when you run scripts. Also because the way `write-source-files` works you need to put them in a special predeclared spot

Reply:
To be more specific on that ^, that is if you want it to be run as an automatic test and verify the output. You can certainly `bazel run //<whatever>` and have it work without that, but if you want it be treated a test (which I think you do so it can all happen during a build) and auto run with `bazel test //...` you need to do this.